streaming/src/main/scala/org/apache/spark/streaming/dstream/InputDStream.scala

```diff
@@ -47,6 +49,27 @@ abstract class InputDStream[T: ClassTag] (@transient ssc_ : StreamingContext)
   /** This is an unique identifier for the input stream. */
   val id = ssc.getNewInputStreamId()
 
+  /**
+   * A rate estimator configured by the user to compute a dynamic ingestion bound for this stream.
+   * @see `RateEstimator`
+   */
+  protected [streaming] val rateEstimator = newEstimator()
+
+  /**
+   * Return the configured estimator, or `noop` if none was specified.
+   */
+  private def newEstimator() =
+    ssc.conf.get("spark.streaming.RateEstimator", "noop") match {
+      case "noop" => new NoopRateEstimator()
+      case estimator => throw new IllegalArgumentException(s"Unknown rate estimator: $estimator")
+    }
+
+
+  // Keep track of the freshest rate for this stream using the rateEstimator
+  protected[streaming] val rateController: RateController = new RateController(id, rateEstimator) {
```

Comment:
I think the rateController should be a option. By default, it will None. Only when backpressure is enabled do we populate it.